### ISSUE TITLE
Confusing warning about `broker_connection_retry_on_startup`

### ISSUE THREAD

**Issue by serl** [celery/celery#9225]
# Checklist

- [x] I have checked the [issues list](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Category%3A+Documentation%22+)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/main)
  to find out if the bug was already fixed in the main branch.
- [x] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Description

We just update Celery and we get this warning:

```
python-3.11/lib/python3.11/site-packages/celery/worker/consumer/consumer.py:508: 
CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
```

Since in our configuration we didn't touch either (`broker_connection_retry` nor `broker_connection_retry_on_startup`), it took me some time to understand what is going on.

Also, from the message it looks like that the default for `broker_connection_retry_on_startup` will be `False`, but the [docs](https://docs.celeryq.dev/en/stable/userguide/configuration.html#broker-connection-retry-on-startup) specify that's enabled by default.

# Suggestions

Assuming that the default in Celery 6 will be to retry every time (both `True` by default), I think that the deprecation warning should not appear if `broker_connection_retry is True`, as no change would result from the deprecation going into action.

To recap, that's my proposition:
- `broker_connection_retry` unset or `True`, `broker_connection_retry_on_startup` unset: no warning (everything evaluated as `True`) <= suggested change
- `broker_connection_retry=False`, `broker_connection_retry_on_startup` unset: warning as it is now
- both `broker_connection_retry` and `broker_connection_retry_on_startup` set: no warning, as it is now

Let me know what are your thoughts, I can prepare a little PR if you agree.

**Comment by Nusnus** [comment]
@thedrow

**Comment by thedrow** [comment]
I think that this is acceptable. Please submit a PR for our review :)
